src/Symfony/Component/Security/Core/Exception/AccessDeniedException.php

```diff
@@ -57,4 +62,26 @@ public function setSubject($subject)
     {
         $this->subject = $subject;
     }
+
+    /**
+     * Sets an access decision and appends the denied reasons to the exception message.
+     */
+    public function setAccessDecision(AccessDecision $accessDecision)
+    {
+        $this->accessDecision = $accessDecision;
+        if (!$accessDecision->getDeniedVotes()) {
+            return;
+        }
+
+        $messages = array_map(static function (Vote $vote) {
+            return $vote->getMessage();
+        }, $accessDecision->getDeniedVotes());
+
+        $this->message .= ' '.rtrim(' '.implode(' ', $messages));
+    }
+
+    public function getAccessDecision(): AccessDecision
```

Comment:
This will error if `setAccessDecision` was not called
```suggestion
    public function getAccessDecision(): ?AccessDecision
```

the property docblock needs to be updated accordingly